llvm/lib/Target/AMDGPU/SIWholeQuadMode.cpp

```diff
@@ -1527,13 +1521,20 @@ bool SIWholeQuadMode::lowerCopyInstrs() {
   for (MachineInstr *MI : LowerToCopyInstrs) {
     LLVM_DEBUG(dbgs() << "simplify: " << *MI);
 
-    Register RecomputeReg = 0;
-    if (MI->getOpcode() == AMDGPU::V_SET_INACTIVE_B32 ||
-        MI->getOpcode() == AMDGPU::V_SET_INACTIVE_B64) {
-      assert(MI->getNumExplicitOperands() == 3);
-      if (MI->getOperand(2).isReg())
-        RecomputeReg = MI->getOperand(2).getReg();
-      MI->removeOperand(2);
+    if (MI->getOpcode() == AMDGPU::V_SET_INACTIVE_B32) {
+      assert(MI->getNumExplicitOperands() == 6);
+
+      LiveInterval *LI = nullptr;
+      if (MI->getOperand(4).isReg())
+        LI = &LIS->getInterval(MI->getOperand(4).getReg());
+
+      MI->removeOperand(5);
+      MI->removeOperand(4);
+      MI->removeOperand(3);
+      MI->removeOperand(1);
+
+      if (LI)
+        LIS->shrinkToUses(LI);
```

Comment:
Using shrinkToUses instead of recomputing the whole interval is just a cleanup. I could commit that separately if it looks OK.

Reply:
Either way is fine by me.

Reply:
#108171